native/exe/SageLauncher/launcher.cpp

```diff
@@ -178,20 +186,24 @@ void popupExceptionError(JNIEnv* env, jthrowable thrower, char* errTitle, HWND h
 	jmethodID toStr = env->GetMethodID(env->GetObjectClass(thrower), "toString", "()Ljava/lang/String;");
 	jstring throwStr = (jstring)env->CallObjectMethod(thrower, toStr);
 	const char* cThrowStr = env->GetStringUTFChars(throwStr, 0);
-	char *errStr = (char*) malloc(env->GetStringLength(throwStr) + 64);
-	sprintf(errStr, "An exception occured in Java:\n%s", cThrowStr);
-	errorMsg(errStr, errTitle, hWnd);
-	env->ReleaseStringUTFChars(throwStr, cThrowStr);
-	free(errStr);
+	size_t sizeerrStr = (env->GetStringLength(throwStr) + 64);
+	char *errStr = (char*)malloc(sizeerrStr);
+	if (errStr != NULL)
+	{
+		sprintf_s(errStr, sizeerrStr, "An exception occured in Java:\n%s", cThrowStr);
+		errorMsg(errStr, errTitle, hWnd);
+		env->ReleaseStringUTFChars(throwStr, cThrowStr);
+		free(errStr);
+	}
 }
 
 jint JNICALL my_vfprintf(FILE *fp, const char *format, va_list args)
 {
 	if (stdOutHandle)
 	{
-	    char buf[1024];
+	    char buf[1024] = {0};
 		DWORD numWrit;
-		DWORD len = _vsnprintf(buf, sizeof(buf), format, args);
+		DWORD len = _vsnprintf_s(buf, sizeof(buf), (sizeof(buf) - strlen(buf)), format, args);

```

Comment:
The strlen(buf) call is unnecessary, the array is initialized two lines up to be empty, so this will always be zero.